### PR TITLE
APIs to define mTLS configuration for KES

### PR DESCRIPTION
**Pull request by Alevsk**
- Adding support for user to define KES mTLS configuration for Vault and Gemalto
- Upgraded to Operator `Version 3.0.10`

### CODE REVIEW

File: swagger.yml [hunk 1]
```diff
@@ -1968,6 +1966,15 @@ definitions:
           ping:
             type: integer
             format: int64
+      tls:
```
Comment (dvaldivia) [inline swagger.yml:1969]: you added these but I don't see them being used anywhere o_O

File: restapi/admin_tenants.go [hunk 2]
```diff
@@ -530,217 +530,275 @@ func getTenantCreatedResponse(session *models.Principal, params admin_api.Create
...
+			Image:    "minio/kes:latest",
+			Replicas: 1,
+			Metadata: nil,
+		}
+		// Using custom image for KES
+		if tenantReq.Encryption.Image != "" {
+			minInst.Spec.KES.Image = tenantReq.Encryption.Image
+		}
+		// Secret to store KES server TLS mTLSCertificates
+		// TODO check if AutoCert it's already configured
```
Comment (dvaldivia) [inline restapi/admin_tenants.go:544]: about this TODO, you need to check it beyond `tenantReq.Autocert==true` ?

File: restapi/admin_tenants.go [hunk 3]
```diff
@@ -530,217 +530,275 @@ func getTenantCreatedResponse(session *models.Principal, params admin_api.Create
...
-				// Initialize AWS
-				kesConfig.Keys.Aws = &kes.Aws{
-					SecretsManager: &kes.AwsSecretManager{},
+				if vaultTLSConfig.Key != "" {
+					clientKey, err := base64.StdEncoding.DecodeString(vaultTLSConfig.Key)
+					if err != nil {
+						return nil, err
+					}
+					mTLSCertificates["client.key"] = clientKey
+					kesConfig.Keys.Vault.TLS.KeyPath = mTLSClientKeyPath
```
Comment (Alevsk) [inline restapi/admin_tenants.go:695]: @dvaldivia the new tls field for vault is being used here

**Comment by Alevsk** [comment]
> could you please Abstract code, also we need to delete all the secrets that are being created if it fails in any of the steps.

ill abstract and refactor the code

File: restapi/admin_tenants.go (hunk 2 above)
Comment (Alevsk) [reply restapi/admin_tenants.go]: fixed

**Pull request merged**